sopel/coretasks.py

```diff
@@ -135,19 +135,31 @@ def startup(bot, trigger):
 
     bot.memory['retry_join'] = dict()
 
+    channels = bot.config.core.channels
     if bot.config.core.throttle_join:
         throttle_rate = int(bot.config.core.throttle_join)
         throttle_wait = max(bot.config.core.throttle_wait, 1)
         channels_joined = 0
-        for channel in bot.config.core.channels:
+
+        LOGGER.info(
+            'Joining %d channels, this may take a moment. '
+            '(throttle JOIN is ON)',
```

Comment:
For this and the below "OFF" variant, I propose this tweaked wording (you can wrap it to pep8 line lengths at will):

```text
Joining %d channels (with JOIN throttle ON); this may take a moment.
```